File: detect_mismatch.py
import re

file_path = 'c:/laragon/www/pontage/backend/modules/salaries.php'
with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
    lines = f.readlines()

last_prepare = ""
last_prepare_line = 0
prepare_param_count = 0

for i, line in enumerate(lines):
    line = line.strip()
    
    # Match ->prepare("...") or ->prepare('...')
    m1 = re.search(r'->prepare\(\s*["\'](.*?)(["\']\s*\))', line)
    m2 = re.search(r'->prepare\(\s*["\'](.*?)["\']', line)
    m = m1 or m2
    if m:
        last_prepare = m.group(1)
        last_prepare_line = i + 1
        prepare_param_count = last_prepare.count('?')
        
    m_exec = re.search(r'->execute\(\[(.*?)\]\)', line)
    if m_exec:
        execute_args = m_exec.group(1).strip()
        execute_param_count = 0
        if execute_args:
            execute_param_count = execute_args.count(',') + 1
            
        if prepare_param_count != execute_param_count:
            print(f"Mismatch at line {i + 1} (prepare at line {last_prepare_line}):")
            print(f"Prepare ({prepare_param_count} params): {last_prepare}")
            print(f"Execute ({execute_param_count} params): {line}")
            print("-" * 60)

File: A SUPPRIMER PLUS TARD/Divers/detect_mismatch.php
<?php
$file = 'c:/laragon/www/pontage/backend/modules/salaries.php';
$content = file_get_contents($file);
$lines = explode("\n", $content);

$last_prepare = "";
$last_prepare_line = 0;
$prepare_param_count = 0;

for ($i = 0; $i < count($lines); $i++) {
    $line = trim($lines[$i]);
    
    if (preg_match('/->prepare\(\s*["\'](.*?)(["\']\s*\))/', $line, $matches) || preg_match('/->prepare\(\s*["\'](.*?)["\']/i', $line, $matches)) {
        $last_prepare = $matches[1];
        $last_prepare_line = $i + 1;
        $prepare_param_count = substr_count($last_prepare, '?');
    }
    
    if (preg_match('/->execute\(\[(.*?)\]\)/', $line, $matches)) {
        $execute_args = $matches[1];
        $execute_param_count = 0;
        if (trim($execute_args) !== "") {
            $execute_param_count = substr_count($execute_args, ',') + 1;
        }
        
        if ($prepare_param_count !== $execute_param_count) {
            echo "Mismatch at line " . ($i + 1) . " (prepare at line $last_prepare_line):\n";
            echo "Prepare ($prepare_param_count params): $last_prepare\n";
            echo "Execute ($execute_param_count params): " . $line . "\n";
            echo "--------------------------------------------------------\n";
        }
    }
}
